src/main/java/controllers/GameSelector.java

```diff
@@ -58,26 +58,49 @@ public void run() {
 
             int sel = this.selectorInput.getUserSelection();
 
-            List<String> usernames = new ArrayList<>();
-
-            String username = this.selectorInput.getUsername();
+            if (sel == 0) {
+                return;
+            }
 
-            while(!username.equalsIgnoreCase("done")) {
-                usernames.add(username);
-                username = this.selectorInput.getUsername();
+            while (!checkValidity(sel)) {
+                this.selectorOutput.sendOutput("Invalid menu selection.\n");
+                sel = this.selectorInput.getUserSelection();
             }
 
+            String gameString = this.games[sel - 1];
+
+            List<String> usernames = new ArrayList<>();
+
             UserManager userManager = new UserManager();
 
-            // check for exit case
-            if (sel == 0) {
-                return;
+            int maxPlayers = GameTemplate.GetMaxPlayers(gameString);
+
+            System.out.println("Input up to " + maxPlayers + " usernames for " +
+                    "players playing the game. Enter 'done' to finish.");
+
+            String username = this.selectorInput.getUsername();
+
+            while (true) {
+                if (username.equalsIgnoreCase("done")) {
+                    System.out.println("\nPlease enter at least one username!\n");
+                    username = this.selectorInput.getUsername();
+                } else {
+                    break;
+                }
             }
 
-            while (!handleUserSelection(sel, usernames, userManager)) {
-                this.selectorOutput.sendOutput("Invalid menu selection.\n");
-                sel = this.selectorInput.getUserSelection();
+            while (!username.equalsIgnoreCase("done") && maxPlayers != 0) {
+                usernames.add(username);
+                try {
+                    userManager.addUser(username);
+                } catch (UserManager.UserAlreadyExistsException e) {
+                    e.printStackTrace();
+                }
+                username = this.selectorInput.getUsername();
+                maxPlayers--;
             }
```

Comment:
When selecting the Crazy Eights game, the user can enter 6 users, but the game only uses 5. 

```
=======================================
              GAME SELECT              
=======================================
[1] Crazy Eights
[0] EXIT
=======================================
Make a selection: 2
Invalid menu selection.
Make a selection: 2
Invalid menu selection.
Make a selection: 1
Input up to 5 usernames for players playing the game. Enter 'done' to finish.
Enter a Username: a
Enter a Username: b
Enter a Username: c
Enter a Username: d
Enter a Username: e
Enter a Username: f
=======================================




=======================================
             Crazy Eights
=======================================




---------------------------------------
a's Turn
---------------------------------------
Top card: 4D
a's Hand: [3S, 6C, 3H, 10C, 8D]
Do you want to draw a card ("Y", "N"): y

---------------------------------------
b's Turn
---------------------------------------
Top card: 4D
b's Hand: [8S, JH, KC, AS, 9S]
Do you want to draw a card ("Y", "N"): y

---------------------------------------
c's Turn
---------------------------------------
Top card: 4D
c's Hand: [4S, 9C, 7H, JD, 5C]
Do you want to draw a card ("Y", "N"): y

---------------------------------------
d's Turn
---------------------------------------
Top card: 4D
d's Hand: [6S, 2H, 10D, QH, 5H]
Do you want to draw a card ("Y", "N"): y

---------------------------------------
e's Turn
---------------------------------------
Top card: 4D
e's Hand: [8C, KS, AD, 2C, 2D]
Do you want to draw a card ("Y", "N"): y

---------------------------------------
a's Turn
---------------------------------------
Top card: 4D
a's Hand: [3S, 6C, 3H, 10C, 8D, 3C]
Do you want to draw a card ("Y", "N"): y

---------------------------------------
b's Turn
---------------------------------------
Top card: 4D
b's Hand: [8S, JH, KC, AS, 9S, 6H]
Do you want to draw a card ("Y", "N"): 
```